### PR TITLE
Add sleep command and sleep list feature to Developers Guide

### PR DESCRIPTION
**Pull request by DaDevChia**
This pull request adds the sleep command and sleep list features to the Developer Guide. It also includes instructions for manual testing and updates the class diagram for the Sleep and SleepList classes to implement modular design.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -342,13 +342,48 @@ retrieving the `ActivityGoalList` from the database and displaying the goals to
 
 5. **Result Display**: A message is returned post-execution and passed through AthletiCLI to the UI for display to the user.
 
+In particular to demonstrate all parser classes, the following sequence diagram shows how the 'edit-sleep' command works:
```
Comment (skylee03) [inline docs/DeveloperGuide.md:345]:
```suggestion
In particular to demonstrate all parser classes, the following sequence diagram shows how the `edit-sleep` command works:
```

**Review by skylee03 — COMMENTED** [review]
You mistakenly removed some of the test cases for misc commands.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DaDevChia) [reply docs/DeveloperGuide.md]: apologies just restored the misc commands

**Review by skylee03 — APPROVED** [review]
LGTM!

**Pull request merged**